swan-lake/development-tutorials/deployment-guide/virtual-machine-deployment.md

```diff
@@ -15,64 +25,155 @@ intro: If you prefer to execute Ballerina services directly on virtual machines
+
+## Centralized deployment
+
+While non-centralized deployment offers simplicity, managing numerous Ballerina artifacts and their dependencies can become complex.  It may also lead to increased resource consumption due to the separate deployment of each artifact.  Centralized deployment addresses these challenges by consolidating all Ballerina artifacts into a single, deployable package, offering a more streamlined and resource-efficient approach.  This strategy is well-suited for scenarios involving multiple Ballerina artifacts that need to be deployed together, or where a more efficient deployment process is desired.
+
+The centralized deployment involves two primary repositories:
+
+- **Source Repository (CI):** Manages Ballerina packages, builds them, and publishes them to Ballerina Central.
+
+- **Deployment Repository (CD):** Consolidates packages from source repositories and deploys the application to the target environment.
+
+### Source Repository
+
+Typically, a single integration can consist of multiple components, each implemented as a separate Ballerina package. These components can represent distinct functionalities or services that collectively form the complete integration solution. By organizing the integration into multiple packages, the source repository ensures modularity, reusability, and easier maintenance. Each package can be developed, tested, and published independently, allowing teams to work on different components in parallel while maintaining a clear separation of concerns.
+
+The continuous integration (CI) process of the centralized deployment is managed by the source repositories.
+
+The following steps outline the CI process in a source repository:
+
+1. Prepare the Server Environment by provisioning the VM or Bare Metal Server.
+   - Ensure the server meets the hardware requirements for your application (CPU, memory, disk space, etc.).
+   - Configure the server OS (Linux is recommended for production).
+
+2. Install the Ballerina distribution from the [Ballerina Installation options.](/downloads/)
+
+3. Install any other dependencies your Ballerina application might need (e.g., database drivers, libraries).
+
+4. Create a new Ballerina package.
+
+   ```bash
+   $ bal new --template lib <package-name>
+   ```
+
+5. Implement the business logic in the Ballerina package.
+
+    > **Tip:**
+    > - Utilize the [Ballerina HTTP default listener](/spec/http/#213-default-listener) to efficiently manage multiple HTTP services on a single port, reducing resource consumption.
+    > - Follow the standard Ballerina project structure, with a `Ballerina.toml` file and source organized in modules if needed. Visit the [Organize Ballerina code](/learn/organize-ballerina-code/) for detailed information about packages and package management.
+
+6. Generate the Ballerina archive.
+
+    ```bash
+    $ bal pack
+    ```
+
+7. Push the package to the Ballerina Central repository.
```

Comment:
Again, the automation of building, testing and publishing is not clear.

Reply:
Fixed via [4c1f3f0](https://github.com/ballerina-platform/ballerina-dev-website/pull/9800/commits/4c1f3f0911c65709821f49e7f6ae08d244becf46)